imucapture/src/main.cpp

```diff
@@ -76,245 +164,178 @@ int kbhit (){
-
-					// fstreaming to a text file
-					{
-						std::ofstream textfileoutput;
-						textfileoutput.open("IMU.txt", std::ofstream::out | std::ofstream::app);
-						// Titles (first line)
-						//textfileoutput << "w,x,y,z,Roll,Pitch,Yaw,Time" << std::endl;
-
-						// Titles (second line)
-						//textfileoutput << "Lat,Lon,Alt,Vel_N,Vel_E,Vel_D,w,x,y,z,Roll,Pitch,Yaw,Time" << std::endl;
-
-						// lat/lon/alt
-						auto position = packet.positionLLA();
-						textfileoutput << std::setw(15) << std::fixed << std::setprecision(6) << std::showpoint << std::right << position[0]
-							<< std::setw(15) << std::fixed << std::setprecision(6) << std::showpoint << std::right << position[1]
-							<< std::setw(15) << std::fixed << std::setprecision(6) << std::showpoint << std::right << position[2]
-							;
-
-						// Get the quaternion data
-						auto quaternion = packet.orientationQuaternion();
-						textfileoutput << "\r"
-							<< "" << std::setw(15) << std::right << std::setprecision(5) << std::showpoint << quaternion.w()
-							<< "" << std::setw(15) << std::right << std::setprecision(5) << std::showpoint << quaternion.x()
-							<< "" << std::setw(15) << std::right << std::setprecision(5) << std::showpoint << quaternion.y()
-							<< "" << std::setw(15) << std::right << std::setprecision(5) << std::showpoint << quaternion.z()
-							;
-
-						// Convert packet to euler
-						auto euler = packet.orientationEuler();
-						textfileoutput << "" << std::setw(15) << std::fixed << std::showpoint << std::right << std::setprecision(5) << euler.roll()
-							<< "" << std::setw(15) << std::fixed << std::showpoint << std::right << std::setprecision(5) << euler.pitch()
-							<< "" << std::setw(15) << std::fixed << std::showpoint << std::right << std::setprecision(5) << euler.yaw()
-							;
-
-						// Get the Timestamp
-						textfileoutput << " " << std::setw(20) << std::fixed << std::showpoint << std::right << XsTime_timeStampNow(0) << "\n"
-							;
-
-
-						//velocity
-						//XsVector3 velocity = packet.velocity();
-						//textfileoutput << std::setw(15) << std::fixed << std::setprecision(3) << std::showpoint << std::right << velocity[0]
-						//	<< std::setw(15) << std::fixed << std::setprecision(3) << std::showpoint << std::right << velocity[1]
-						//	<< std::setw(15) << std::fixed << std::setprecision(3) << std::showpoint << std::right << velocity[2]
-						//	;
-
-
-						textfileoutput.close();
-
-					}
-
-					std::cout << std::flush;
-				}
-				msgs.clear();
-				XsTime::msleep(0);
-			}
-			std::cin.get();
-			std::cout << std::endl << std::string(79, '-') << std::endl;
-		}catch (std::runtime_error const & error){
-			std::cout << error.what() << std::endl;
-		}catch (...){
-			std::cout << "An unknown fatal error has occured. Aborting." << std::endl;
-		}
+inline void
+print_position_to_file(std::ofstream &outfile, XsDataPacket &packet)
+{
+	XsVector position = packet.positionLLA();
```

Comment:
Could be auto?

Reply:
Could be, but I think these ones we should leave alone because not all of the data coming from the packets are vectors, and keeping the type can just keep it clearer.